lib/rubocop/cop/rspec/change_by_zero.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Prefer negated matchers over `to change.by(0)`.
+      #
+      # @example
+      #   # bad
+      #   expect { run }.to change(Foo, :bar).by(0)
+      #   expect { run }.to change { Foo.bar }.by(0)
+      #   expect { run }
+      #     .to change(Foo, :bar).by(0)
+      #     .and change(Foo, :baz).by(0)
+      #   expect { run }
+      #     .to change { Foo.bar }.by(0)
+      #     .and change { Foo.baz }.by(0)
+      #
+      #   # good
+      #   expect { run }.not_to change(Foo, :bar)
+      #   expect { run }.not_to change { Foo.bar }
+      #   expect { run }
+      #     .to not_change(Foo, :bar)
+      #     .and not_change(Foo, :baz)
+      #   expect { run }
+      #     .to not_change { Foo.bar }
+      #     .and not_change { Foo.baz }
+      #
+      class ChangeByZero < Base
+        extend AutoCorrector
+        MSG = 'Prefer `not_to change` over `to change.by(0)`.'
+        MSG_COMPOUND = 'Prefer negated matchers with compound expectations ' \
+                          'over `change.by(0)`.'
+
+        # @!method expect_change_with_arguments(node)
+        def_node_matcher :expect_change_with_arguments, <<-PATTERN
+          (send
+            (send nil? :change ...) :by
+            (int 0))
+        PATTERN
+
+        # @!method expect_change_with_block(node)
+        def_node_matcher :expect_change_with_block, <<-PATTERN
+          (send
+            (block
+              (send nil? :change)
+              (args)
+              (send (...) $_)) :by
+            (int 0))
+        PATTERN
+
+        def on_send(node)
+          expect_change_with_arguments(node) do
```

Comment:
minor: the cop will be faster if we RESTRICT_ON_SEND to `change` and then check the pattern against `node.parent` (as we will be getting the inner send node)

Reply:
> the cop will be faster if we RESTRICT_ON_SEND to […]

Can we write a cop to verify the presence of RESTRICT_ON_SEND when `on_send` is defined?

Reply:
Thank you so much. I'll define RESTRICT_ON_SEND to `change`.

Reply:
> Can we write a cop to verify the presence of RESTRICT_ON_SEND when on_send is defined?

I tried to create a PR https://github.com/rubocop/rubocop/pull/10783 but it is now closed.
The reason is that there was a judgment that there may be more noise than detection efficiency.